src/lightning/data/streaming/shuffle.py

```diff
@@ -144,5 +144,5 @@ def get_chunks_and_intervals_per_ranks(self, distributed_env: _DistributedEnv, c
 
         return chunks_per_ranks, intervals_per_ranks
 
-    def __call__(self, array: np.ndarray, current_epoch: int, chunk_index: int) -> List[int]:
-        return np.random.RandomState(seed=self.seed + current_epoch + chunk_index).permutation(array).tolist()
+    def __call__(self, array: np.ndarray, epoch_shift: int, chunk_index: int) -> List[int]:
+        return np.random.RandomState(seed=self.seed + epoch_shift + chunk_index).permutation(array).tolist()
```

Comment:
renaming the variable won't help. My comment still applies. The random state will be the same if the sum of  (epoch_shift + chunk_index) is the same for different values of epoch_shift and chunk_index

Reply:
How about you just pass the sequence to the random state directly?

```suggestion
        return np.random.RandomState([self.seed, current_epoch, chunk_index]).permutation(array).tolist()
```

Reply:
But now, it should be unique for every chunk every epoch.

Reply:
As shift_epoch is `num_chunks * current_epoch`.

Reply:
Outside the function yes, but inside the function shift_epoch and index can be values that chosen such that they give the same value. It shouldn't be the responsibility of the caller to know the implementation detail. So my suggestion is to either move your change into the function or use the sequence as input to the random state.

Reply:
Yes, you want this to be deterministic for a given epoch_shift and chunk_index.

Reply:
This also simply random seed handling where we don't need to keep track of it nor to store it.

Reply:
What do you think @awaelchli ?